### PR TITLE
Add ability to request user from token refresh

### PR DESCRIPTION
**Pull request by ph0t0shop**
This is needed to be able to refresh profile information into a launcher_profiles.json, something for which I will make a PR soon in node-minecraft-protocol

### CODE REVIEW

**Review by wvffle — APPROVED** [review]

**Comment by DeltaEvo** [comment]
You need to update the refresh test https://github.com/PrismarineJS/node-yggdrasil/blob/master/test/index.js#L133-L163

**Comment by rom1504** [comment]
@ph0t0shop could you please fix the test ?

**Comment by ph0t0shop** [comment]
@rom1504 I remember them already being correct, this was probably the CI bug that was here before. an empty commit triggered passing tests

File: lib/Client.js [hunk 1]
```diff
@@ -42,11 +42,12 @@ Client.auth = function (options, cb) {
  * @param  {String}   client Client Token
  * @param  {Function} cb     (err, new token, full response body)
  */
-Client.refresh = function (access, client, cb) {
+Client.refresh = function (access, client, cb, requestUser) {
   const host = this.host || defaultHost
```
Comment (rom1504) [inline lib/Client.js:46]: Callback should always be the last argument.
If you want to have an optional argument you can use the trick to check the argument type

Comment (ph0t0shop) [reply lib/Client.js:46]: I did this for backwards compatible reasons. Is it okay to change the order?

Comment (rom1504) [reply lib/Client.js:46]: If the parameter is not optional then it's a breaking change regardless of the order
If the parameter is optional, you can use the trick I mentioned to keep the callback at the end without making a breaking change
So I'd advise the second option

**Comment by rom1504** [comment]
Can you update the doc ?

**Review by rom1504 — APPROVED** [review]

**Pull request merged**